index.tsx

```diff
@@ -732,6 +737,36 @@ function anyScopeRegex(regex: RegExp) {
   }
 }
 
+function matchParagraph(source: string, state: MarkdownToJSX.State, prevCapturedString?: string) {
```

Comment:
I'm replacing the regex with imperative code since:
1. I found the current PARAGRAPH_R regex hard to reason about (and hard to debug) as is
2. In #291 @probablyup said "there must be a simpler way to do this than jamming in all the other syntax detection regex", and I agree it doesn't make it easier, which is why I'm imperatively testing the other block regex rather than "jamming them in".

Reply:
Will review this week, thank you for taking this on

Reply:
👋